FluentUI/src/main/java/com/microsoft/fluentui/contextualcommandbar/ContextualCommandBar.kt

```diff
@@ -0,0 +1,262 @@
+package com.microsoft.fluentui.contextualcommandbar
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.graphics.drawable.GradientDrawable
+import android.support.annotation.DrawableRes
+import android.support.v4.content.ContextCompat
+import android.support.v7.content.res.AppCompatResources
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.HorizontalScrollView
+import android.widget.ImageView
+import android.widget.LinearLayout
+import com.microsoft.fluentui.R
+import com.microsoft.fluentui.util.ThemeUtil
+import com.microsoft.fluentui.util.isVisible
+
+class ContextualCommandBar @JvmOverloads constructor(
+        context: Context,
+        attrs: AttributeSet? = null,
+        defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+    private var horizontalScrollView = HorizontalScrollView(context)
+    private var commandContainer = LinearLayout(context)
+
+    // Layout configuration
+    private var groupSpace = 0
+    private var itemSpace = 0
+    private var itemPaddingVertical = 0
+    private var itemPaddingHorizontal = 0
+
+    // Dismiss button configuration
+    var showDismiss = false
+        set(value) {
+            field = value
+            setDismissButtonVisible(value)
+        }
+
+    @DrawableRes
+    var dismissIcon: Int = 0
+        set(value) {
+            if (value == 0) {
+                return
+            }
+
+            field = value
+            dismissButton?.setImageResource(dismissIcon)
+        }
+    private var dismissButton: ImageView? = null
+    private var dismissButtonContainer: LinearLayout? = null
+
+    var dismissListener: (() -> Unit)? = null
+    var itemClickListener: OnItemClickListener? = null
+
+    init {
+        removeAllViews()
+        horizontalScrollView.removeAllViews()
```

Comment:
When is this required ? I See a new instance of horizontalScrollView here and no child views attached